.github/workflows/terraform_validate.yaml

```diff
@@ -5,21 +5,23 @@ on:
 
 jobs:
   tf_validate:
-    name: 'Terraform validate'
+    name: 'Terraform validate (on tf ${{ matrix.terraform_version }})'
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        terraform_version: ['~1.1.0', '~1.2.0', '~1.3.0', '~1.4.0', '~1.5.0', '~1.6.0', '~1.7.0', 'latest']
```

Comment:
Ok, my only question is if we don't go < 1.4.0 for integration test, why doing it for tf validate here? I'd put a comment here as same as we have in rest of actions

Reply:
validation itself will catch things (like `move` keyword, introduced in 1.1) - so better than nothing.  

integration is slower and more brittle (if AWS is slow to reflect state of changes); and in practice oldest customer terraform edition I've seen is 1.4.7 recently, and current release is 1.7 with 1.8 in beta now.

i also think terraform configs are something people leave in place for **years**, so we need to be pretty generous about range of compatibility